flocker/control/_diffing.py

```diff
@@ -76,7 +79,9 @@ class _Set(PClass):
     value = field()
 
     def apply(self, obj):
-        return obj.transform(self.path, self.value)
+        return obj.transform(
+            self.path[:-1], lambda o: o.set(self.path[-1], self.value)
```

Comment:
I've added new interfaces which hopefully describe the API of the object that will be supplied to the transform operation.